gateway/gateway.go

```diff
@@ -586,14 +583,42 @@ func (gateway *HandleT) beaconBatchHandler(w http.ResponseWriter, r *http.Reques
 	gateway.beaconHandler(w, r, "batch")
 }
 
+func (gateway *HandleT) checkAndValidateWebHandler(w http.ResponseWriter, r *http.Request, reqType string) string {
+	var sourceFailStats = make(map[string]int)
+	var errorMessage = ""
+	writeKey, _, ok := r.BasicAuth()
+	if !ok || writeKey == "" {
+		errorMessage = response.GetStatus(response.NoWriteKeyInBasicAuth)
+		misc.IncrementMapByKey(sourceFailStats, "noWriteKey", 1)
+		gateway.updateSourceStats(sourceFailStats, "gateway.write_key_failed_requests")
+		return errorMessage
+	}
+	payload, err := gateway.getPayloadFromRequest(r)
+	if err == nil {
+		if reqType == "import" {
+			errorMessage = gateway.userWebImportHandler(w, r, "batch", payload, writeKey)
+		} else {
+			done := make(chan string, 1)
+			gateway.AddToWebRequestQ(r, &w, done, reqType, payload, writeKey)
+			//Wait for batcher process to be done
+			errorMessage = <-done
+		}
+	} else if err.Error() == "RequestBodyNil" {
```

Comment:
Don't we want to collect any stats in this case?

Reply:
Add Source Fail Stats to update source stats